Sources/DispatchAsync/DispatchGroup.swift

```diff
@@ -0,0 +1,131 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2025 Apple Inc. and the Swift.org project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of Swift.org project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+// MARK: - Public Interface for Non-Async Usage -
+
+/// `DispatchGroup` is a drop-in replacement for the `DispatchGroup` implemented
+/// in Grand Central Dispatch. However, this class uses Swift Concurrency, instead of low-level threading API's.
+///
+/// The primary goal of this implementation is to enable WASM support for Dispatch.
+///
+/// Refer to documentation for the original [DispatchGroup](https://developer.apple.com/documentation/dispatch/dispatchgroup)
+/// for more details,
+@available(macOS 10.15, *)
+public class DispatchGroup: @unchecked Sendable {
+    /// Used to ensure FIFO access to the enter and leave calls
+    @globalActor
```

Comment:
How does forcing this onto the global actor impact separately instantiated DispatchGroups? It seems like it would force all of them to execute sequentially, interleaving the results. i.e. Not only is this FIFO within the group, it looks like it's globally FIFO.

If this is the intended behavior, given that WASM is single-threaded, could we enforce or document that to avoid any expectation that this would operate in a multithreaded swift-concurrency environment?